### PR TITLE
Fix up: pcie slots not enough for devices attaching

### PR DESCRIPTION
**Pull request by chloerh**
Add pcie slots to vm before attaching device to avoid errors like
"Not enough slots"

Signed-off-by: Haijiao Zhao <haizhao@redhat.com>

### CODE REVIEW

**Comment by chloerh** [comment]
PASS 1-type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.scalability_test.max_vfs.nop.vf.managed_yes.driver.ixgbe

**Review by smitterl — COMMENTED** [review]
@chloerh I'm wondering if it is possible and if it would make sense to add the logic to add more slots in a unified way to `avocado-vt`?

I see that this issue of not having enough slots has been happening in several test suites:
PR search - https://github.com/autotest/tp-libvirt/pulls?q=is%3Apr+pci+slots+is%3Aclosed
leading to several in-test fixes, e.g.
https://github.com/autotest/tp-libvirt/pull/2650
https://github.com/autotest/tp-libvirt/pull/2692
https://github.com/autotest/tp-libvirt/pull/2780
https://github.com/autotest/tp-libvirt/pull/2868
https://github.com/autotest/tp-libvirt/pull/2990
...

**Comment by chloerh** [comment]
Oh I see, I agree with you.
@Dan Zheng <dzheng@redhat.com> @Chunfu Wen <chwen@redhat.com> @Kyla Zhang
<weizhan@redhat.com> @Yingshun Cui <yicui@redhat.com>
Would you like to discuss this issue, to find a solution?

Thanks,
Haijiao

File: libvirt/tests/src/sriov/sriov.py [hunk 1]
```diff
@@ -750,6 +749,20 @@ def remove_devices(vmxml, device_type):
                     count += 1
                 if max_vfs_attached:
                     interface_list = []
+                    # Get max index of all pcie-root-port
+                    pcie_ctls = vmxml.get_controllers('pci', 'pcie-root-port')
+                    pcie_indexes = [int(port.get('index')) for port in pcie_ctls]
+
+                    def _add_pcie_root_port(index):
```
Comment (kylazhang) [inline libvirt/tests/src/sriov/sriov.py:756]: Please give docstring for the func, thanks

File: libvirt/tests/src/sriov/sriov.py [hunk 2]
```diff
@@ -750,6 +749,20 @@ def remove_devices(vmxml, device_type):
...
+                        pcie_root_port = Controller("controller")
+                        pcie_root_port.type = "pci"
+                        pcie_root_port.index = index
+                        pcie_root_port.model = "pcie-root-port"
+                        vmxml.add_device(pcie_root_port)
+                        vmxml.sync()
```
Comment (kylazhang) [inline libvirt/tests/src/sriov/sriov.py:762]: And blank line needed after func

File: libvirt/tests/src/sriov/sriov.py (hunk 1 above)
Comment (chloerh) [reply libvirt/tests/src/sriov/sriov.py:756]: Updated

File: libvirt/tests/src/sriov/sriov.py (hunk 2 above)
Comment (chloerh) [reply libvirt/tests/src/sriov/sriov.py:767]: Updated

**Review by kylazhang — APPROVED** [review]

**Review by smitterl — APPROVED** [review]
LGTM. Reviewing similar cases and moving logic to avocado-vt shouldn't be blocking this.

**Comment by chloerh** [comment]
Thank you for reviewing, @kylazhang @smitterl .

**Pull request merged**